scipy/integrate/_cubature.py

```diff
@@ -374,3 +469,226 @@ def _process_subregion(data):
     err_sub = rule.estimate_error(f, a_k_sub, b_k_sub, args)
 
     return a_k_sub, b_k_sub, est_sub, err_sub
+
+
+def _is_strictly_in_region(point, a, b):
+    xp = array_namespace(point, a, b)
+
+    if xp.all(point == a) or xp.all(point == b):
+        return False
+
+    return xp.all(a <= point) and xp.all(point <= b)
+
+
+def _split_region_at_points(a, b, points):
+    """
+    Given the integration limits `a` and `b` describing a rectangular region and a list
+    of `points`, find the list of ``[(a_1, b_1), ..., (a_l, b_l)]`` which breaks up the
+    initial region into smaller subregion such that no `points` lie strictly inside
+    any of the subregions.
+    """
+
+    xp = array_namespace(a, b)
+    regions = [(a, b)]
+
+    for point in points:
+        if xp.any(xp.isinf(point)):
+            # If a point is specified at infinity, ignore.
+            #
+            # This case occurs when points are given by the user to avoid, but after
+            # applying a transformation, they are removed.
+            continue
+
+        new_subregions = []
+
+        for a_k, b_k in regions:
+            if _is_strictly_in_region(point, a_k, b_k):
+                subregions = _split_subregion(a_k, b_k, point)
+
+                for left, right in subregions:
+                    # Skip any zero-width regions.
+                    if xp.any(left == right):
+                        continue
+                    else:
+                        new_subregions.append((left, right))
+
+                new_subregions.extend(subregions)
+
+            else:
+                new_subregions.append((a_k, b_k))
+
+        regions = new_subregions
+
+    return regions
+
+
+class _VariableTransform:
```

Comment:
Are there other transforms planned?

Reply:
Yes, integrals with functions as limits are also calculated using a transform. But if you think two transforms aren't enough to justify the base class, then I'm happy to remove this.

Reply:
If there's stuff to come then I think this is fine